### PR TITLE
Upload Button/Progress Bar/Init bootloader process

### PR DESCRIPTION
**Pull request by josemapac2**
+ Adding a progress bar close to the button "Upload" in the "Device Settings/Update Firmware" menu.
+ Making the "Upload" button work to Update the Firmware, by forcing the bootloader process sending a command through the USB serial port. 
+ Fix the bug in the view CONNECT DEVICE to automatically detect all Daqifi USB ports: Check there is no 'null'

### CODE REVIEW

File: Daqifi.Desktop.Setup/UpgradeLog2.htm [hunk 1]
```diff
@@ -0,0 +1,268 @@
+﻿<!DOCTYPE html>
```
Comment (tylerkron) [inline Daqifi.Desktop.Setup/UpgradeLog2.htm:1]: We don't need to check in this file

File: Daqifi.Desktop.Setup/UpgradeLog.htm [hunk 2]
```diff
@@ -0,0 +1,278 @@
+﻿<!DOCTYPE html>
```
Comment (tylerkron) [inline Daqifi.Desktop.Setup/UpgradeLog.htm:1]: Don't need to check in this file

File: Daqifi.Desktop/ViewModels/DaqifiViewModel.cs [hunk 3]
```diff
@@ -48,6 +56,227 @@ public class DaqifiViewModel : ViewModelBase
         private bool _canToggleLogging;
         private string _loggedDataBusyReason;
         private string _firmwareFilePath;
+        //private readonly Pic32Bootloader _bootloader;
```
Comment (tylerkron) [inline Daqifi.Desktop/ViewModels/DaqifiViewModel.cs:59]: Delete this commented line?

File: Daqifi.Desktop/ViewModels/DaqifiViewModel.cs [hunk 4]
```diff
@@ -16,9 +18,15 @@
 using System.Collections.ObjectModel;
 using System.ComponentModel;
 using System.Linq;
+using System.IO;
 using System.Threading.Tasks;
 using System.Windows;
 using System.Windows.Input;
+using System.Collections;
+using System.Diagnostics;
```
Comment (tylerkron) [inline Daqifi.Desktop/ViewModels/DaqifiViewModel.cs:26]: This is unused

File: Daqifi.Desktop/ViewModels/DaqifiViewModel.cs [hunk 5]
```diff
@@ -16,9 +18,15 @@
...
+using Daqifi.Desktop.Device.SerialDevice;
+using System.Threading;
+using System.Windows.Interop;
```
Comment (tylerkron) [inline Daqifi.Desktop/ViewModels/DaqifiViewModel.cs:29]: This is unused

**Review by tylerkron — CHANGES_REQUESTED** [review]
Thanks for the PR! Just a couple of cleanup items I saw!

**Review by cptkoolbeenz — APPROVED** [review]
Using firmware update feature with stock bootloader causes Windows confusion when switching from COM to HID device rapidly.  Adding a 2s delay during bootloader startup solves this issue.

**Pull request merged**